client/resdef.go

```diff
@@ -22,19 +22,31 @@ import (
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/apis/extensions"
 	"k8s.io/kubernetes/pkg/client/restclient"
 )
 
+type ReplicaSet struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// Standard object metadata
+	v1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	ReplicaSet    *extensions.ReplicaSet `json:"replicaset"`
+	SuccessFactor int32                  `json:"success_factor"`
```

Comment:
As we agreed,  we keep SF as a part of RC key and as a part of dependency onto RC. So it should not be duplicated in yaml.